EosPayload/lib/orcheostrator/orcheostrator.py

```diff
@@ -84,64 +97,125 @@ def valid_driver(driver) -> bool:
-            and driver.__name__ != "DriverBase"
+            and driver.__name__ not in base_drivers
         )
 
+    @staticmethod
+    def parse_config(config_path: str) -> dict:
+        """ Parses a config file at the provided path
+
+        :param config_path: the path to the config file
+        :return: a validated config dict
+        """
+        available_drivers = {}
+        for attribute_name in dir(drivers):
+            driver = getattr(drivers, attribute_name)
+            if OrchEOStrator.valid_driver(driver):
+                available_drivers[attribute_name] = driver
+
+
+        with open(config_path) as config_file:
+            config = json.load(config_file)
+
+        used_ids = []
+        enabled_driver_list = []
+
+        for driver_config in config['enabled_drivers']:
+            # Automatically populate name if not provided
+            if driver_config.get("name") is not None:
+                driver_config['name'] = driver_config.get("name")
+            else:
+                # Convert default name from CamelCase to lowercase with dashes
+                # Code snippet taken from https://github.com/jpvanhal/inflection
+                driver_class_name = driver_config.get("driver_class")
+                driver_class_name = re.sub(r"([A-Z]+)([A-Z][a-z])", r'\1_\2', driver_class_name)
+                driver_class_name = re.sub(r"([a-z\d])([A-Z])", r'\1_\2', driver_class_name)
+                driver_class_name = driver_class_name.replace("_", "-")
+                driver_class_name = driver_class_name.lower()
+
+                driver_config['name'] = driver_class_name
+
+            # Validate name
+            driver_name = driver_config.get("name")
+            if not driver_name.isascii() or not driver_name.replace("-", "").isalnum():
+                raise ValueError("Driver names may only contain alphanumeric characters and hyphens.")
+
+            # Validate ID
+            driver_device_id = driver_config.get("device_id")
+            if driver_device_id is None:
+                raise ValueError("Driver ID Cannot be None.")
+            if driver_device_id in used_ids:
+                raise ValueError("Driver ID Must be unique.")
+            try:
+                driver_config["device_id"] = EosLib.Device[driver_config["device_id"]].value
+            except KeyError:
+                raise ValueError("Invalid Device ID.")
+
+            used_ids.append(driver_device_id)
+
+            new_driver_class = driver_config['driver_class']
+            if new_driver_class in available_drivers:
+                for field in available_drivers.get(new_driver_class).get_required_device_config():
+                    if driver_config.get("driver_settings") is None or driver_config.get("driver_settings").get(field) is None:
+                        raise ValueError("Required driver setting not provided.")
+                enabled_driver_list.append((available_drivers[new_driver_class], driver_config))
```

Comment:
why use an intermediate tuple instead of DriverContainer?

Reply:
I think this is a skeleton from before I modified DriverContainer, I'll change it over